app/shared/taginput/taginput.component.ts

```diff
@@ -0,0 +1,156 @@
+﻿import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { User } from '../../core/user/user.model';
+import { UserService } from '../../core/user/user.service';
+
+import {
+  Validators,
+  FormGroup,
+  FormControl
+} from '@angular/forms';
+
+@Component({
+  selector: 'hip-taginput',
+  templateUrl: './app/shared/taginput/taginput.component.html'
+})
+export class TagInputComponent {
+
+  public errorMessage: any;       // Handling error message
+  public names: string[] = [];    // AutoComplete List
+  @Input() role: string;          // User role Passed dynamically
+  @Input() users: number[];         // List of Users updated
```

Comment:
Furthermore if you use the full User, you don't need to change the model and add a Array, which needs to be consistence to another array, so every time we update one array, we have to update the other one as well. 
So it is easier to have a consistence model by using the full User as Input and Output, since the Component who uses TagInput is working with the full User as model.